src/middleware/api.ts

```
@@ -0,0 +1,640 @@
+
+export function createValidationMiddleware<T>(
+  schema: z.ZodSchema<T>,
+  options: { field?: "body" | "query" | "params" } = { field: "body" }
+) {
+  return async (req: NextRequest): Promise<NextRequest> => {
+    try {
+      let dataToValidate;
+
+      switch (options.field) {
+        case "query":
+          const url = new URL(req.url);
```

Biome (2.1.2)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Safe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)